src/main/java/com/faforever/client/os/FileOpeningHandler.java

```diff
@@ -0,0 +1,54 @@
+package com.faforever.client.os;
+
+import com.faforever.client.replay.ReplayService;
+import com.install4j.api.launcher.StartupNotification;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * When a file type is associated with the client and the user opens such a file, this class will handle the opening
+ * event. Since Install4j only allows a single listener, all file types need to be handled in this class.
+ */
+@Component
+@Slf4j
+public class FileOpeningHandler implements ApplicationRunner {
+
+  private final ReplayService replayService;
+
+  public FileOpeningHandler(ReplayService replayService) {
+    this.replayService = replayService;
+  }
+
+  @PostConstruct
+  public void postConstruct() {
+    log.debug("Registering file opening handler: {}", this.getClass().getName());
+    StartupNotification.registerStartupListener(this::onStartup);
+  }
+
+  private void onStartup(String parameters) {
+    log.debug("Handling startup: {}", parameters);
+    if (parameters.split("\" \"").length > 2) {
+      throw new IllegalArgumentException("Can't handle multiple files: " + parameters);
```

Comment:
should you not '{}'?

Reply:
or %s

Reply:
Exceptions don't support string formatting. I had to do:
```throw new IllegalArgumentException(String.format("Can't handle multiple files: %s", parameters));```
but with simple concatenations like this, I don't see the benefit. With more/complex parameters, I would agree.